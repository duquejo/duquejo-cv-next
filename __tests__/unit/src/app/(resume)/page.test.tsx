import MainPage from '@/app/(resume)/page';
import { render, within } from '@testing-library/react';

describe('<MainPage /> tests', () => {
  it('Should match the snapshot', () => {
    const { container } = render(<MainPage />);

    expect(container).toMatchSnapshot();

    const wrapper = within(container);
    expect(wrapper).toBeDefined();

    expect(
      wrapper.getByRole('heading', { level: 1, name: /Hi, I'm Jos√© Duque./ }),
    ).toBeInTheDocument();
    expect(
      wrapper.getByRole('heading', { level: 2, name: 'What have I done?' }),
    ).toBeInTheDocument();
    expect(
      wrapper.getByRole('heading', { level: 2, name: 'Academic studies & Certifications' }),
    ).toBeInTheDocument();
    expect(wrapper.getByRole('heading', { level: 2, name: 'Languages' })).toBeInTheDocument();
  });
});
