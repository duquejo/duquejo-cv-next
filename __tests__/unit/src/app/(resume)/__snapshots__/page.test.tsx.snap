// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<MainPage /> tests > Should match the snapshot 1`] = `
<div>
  <section
    class="flex flex-col lg:flex-row lg:justify-center items-center mx-0 mt-5 py-10 relative"
  >
    <div
      class="order-2 lg:basis-3/5 lg:order-1 px-10 pb-0 pt-5 lg:pt-0"
    >
      <span
        class="text-foreground font-light mt-10 animate-entrance duration-1000"
      >
        Software Engineer
      </span>
      <h1
        class="text-4xl lg:text-6xl mb-5 animate-entrance duration-1000"
      >
        Hi, I'm 
        <strong>
          José Duque.
        </strong>
      </h1>
      <p
        class="text-justify"
      >
        Colombian Software Engineer with more than 8 years of experience in web application development.
      </p>
      <br />
      <p
        class="text-justify"
      >
        I am focused on guaranteeing quality solutions based on good practices with technologies on 
        <b>
          monolithic architectures
        </b>
         as well as 
        <b>
          microservices architectures
        </b>
        .
      </p>
      <div
        class="flex justify-center gap-x-10 mt-8 w-full md:w-auto"
      >
        <img
          alt="Typescript"
          class="hover:animate-wiggle"
          data-nimg="1"
          decoding="async"
          height="50"
          loading="lazy"
          src="/static/svg/typescript.svg"
          style="color: transparent;"
          width="50"
        />
        <img
          alt="Node JS"
          class="hover:animate-wiggle"
          data-nimg="1"
          decoding="async"
          height="50"
          loading="lazy"
          src="/static/svg/nodejs.svg"
          style="color: transparent;"
          width="50"
        />
        <img
          alt="Vue JS"
          class="hover:animate-wiggle"
          data-nimg="1"
          decoding="async"
          height="50"
          loading="lazy"
          src="/static/svg/vue.svg"
          style="color: transparent;"
          width="50"
        />
        <img
          alt="React JS"
          class="hover:animate-wiggle"
          data-nimg="1"
          decoding="async"
          height="50"
          loading="lazy"
          src="/static/svg/react.svg"
          style="color: transparent;"
          width="50"
        />
      </div>
    </div>
    <picture
      class="overflow-hidden relative max-w-md order-1 flex-1 lg:order-2 px-10 lg:mt-0"
    >
      <svg
        class="lucide lucide-music absolute top-0 text-primary/50 animate-bounce-subtle repeat-infinite z-10"
        fill="none"
        height="64"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="3"
        viewBox="0 0 24 24"
        width="64"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9 18V5l12-2v13"
        />
        <circle
          cx="6"
          cy="18"
          r="3"
        />
        <circle
          cx="18"
          cy="16"
          r="3"
        />
      </svg>
      <svg
        class="lucide lucide-gamepad absolute bottom-0 left-7 text-primary/50 animate-wiggle z-10"
        fill="none"
        height="70"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="3"
        viewBox="0 0 24 24"
        width="70"
        xmlns="http://www.w3.org/2000/svg"
      >
        <line
          x1="6"
          x2="10"
          y1="12"
          y2="12"
        />
        <line
          x1="8"
          x2="8"
          y1="10"
          y2="14"
        />
        <line
          x1="15"
          x2="15.01"
          y1="13"
          y2="13"
        />
        <line
          x1="18"
          x2="18.01"
          y1="11"
          y2="11"
        />
        <rect
          height="12"
          rx="2"
          width="20"
          x="2"
          y="6"
        />
      </svg>
      <svg
        class="lucide lucide-loader-circle absolute right-10 text-primary animate-spin duration-slow repeat-infinite z-10"
        fill="none"
        height="64"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="5"
        viewBox="0 0 24 24"
        width="64"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M21 12a9 9 0 1 1-6.219-8.56"
        />
      </svg>
      <svg
        class="lucide lucide-send absolute right-8 bottom-5 text-primary animate-bounce-subtle repeat-infinite z-10"
        fill="none"
        height="50"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="3"
        viewBox="0 0 24 24"
        width="50"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M14.536 21.686a.5.5 0 0 0 .937-.024l6.5-19a.496.496 0 0 0-.635-.635l-19 6.5a.5.5 0 0 0-.024.937l7.93 3.18a2 2 0 0 1 1.112 1.11z"
        />
        <path
          d="m21.854 2.147-10.94 10.939"
        />
      </svg>
      <svg
        class="lucide lucide-circle absolute right-36 top-10 text-primary/50 animate-pulse duration-slow"
        fill="none"
        height="40"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="5"
        viewBox="0 0 24 24"
        width="40"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="12"
          cy="12"
          r="10"
        />
      </svg>
      <svg
        class="lucide lucide-circle absolute left-20 bottom-24 text-primary/50 animate-pulse duration-slow delay-150"
        fill="none"
        height="20"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="5"
        viewBox="0 0 24 24"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="12"
          cy="12"
          r="10"
        />
      </svg>
      <img
        alt="José Miguel Duque"
        class="rounded-full backdrop-blur-sm bg-primary/5 shadow-inner object-cover md:max-w-xs"
        data-nimg="1"
        decoding="async"
        height="950"
        src="/_next/image?url=%2Fstatic%2Fimg%2Favatar_cropped_950.png&w=1920&q=75"
        srcset="/_next/image?url=%2Fstatic%2Fimg%2Favatar_cropped_950.png&w=1080&q=75 1x, /_next/image?url=%2Fstatic%2Fimg%2Favatar_cropped_950.png&w=1920&q=75 2x"
        style="color: transparent;"
        width="950"
      />
    </picture>
  </section>
  <section
    class="m-10 gap-x-10 gap-y-2"
  >
    <h2
      class="text-lg lg:text-xl mb-3 font-semibold underline-offset-4 underline decoration-primary"
    >
      What have I done?
    </h2>
    <p
      class="text-justify"
    >
      During my working life, I have had the opportunity to have a broad view of the client specific needs to solve any technological challenge that arises; I have played roles as advisor and coordinator in development teams for public sector companies such as the
       
      <strong>
        Alcaldía de Itagüí
      </strong>
      , 
      <strong>
        Alcaldía de Bogotá
      </strong>
      ,
       
      <strong>
        Ministerio de Educación Nacional
      </strong>
       and 
      <strong>
        EAFIT University
      </strong>
      . I have also participated in custom development projects and website development with
       
      <strong>
        Marketing agencies
      </strong>
       full-time and alternatively as a
       
      <strong>
        Freelancer
      </strong>
       part-time.
      <br />
      <br />
      With 
      <strong>
        software factories
      </strong>
       I have had the opportunity to work for
       
      <strong>
        Educational Startups
      </strong>
       with great regional reach, as well as for the
       
      <strong>
        Banking
      </strong>
       sector, 
      <strong>
        Insurances
      </strong>
       and 
      <strong>
        Health
      </strong>
      , which have allowed us to expand knowledge and comply with the rigidity of the policies of said market.
      <br />
      <br />
      In accordance with the search for a more specialized orientation to the most popular technologies, I have participated in national intensive learning sessions of
       
      <strong>
        Node JS
      </strong>
       and carried out individual studies on Frontend technologies such as 
      <strong>
        React JS
      </strong>
      , 
      <strong>
         Vue JS
      </strong>
      , as well as other basic approaches in 
      <strong>
        Angular
      </strong>
      .
      <br />
      <br />
      From the Backend, studies in 
      <strong>
        architectures based on microservices
      </strong>
       using 
      <strong>
        Amazon Web Services
      </strong>
       and 
      <strong>
        Azure DevOps
      </strong>
      , this in search of profiling myself as a competent Full Stack developer for the needs of the current market.
    </p>
  </section>
  <section
    class="flex flex-col lg:flex-row gap-x-10 m-10"
  >
    <div
      class="basis-full"
    >
      <h2
        class="main-subtitle"
      >
        Academic studies & Certifications
      </h2>
      <ol
        class="first-of-type:pt-0 pt-5 relative border-s border-gray-200"
      >
        <div
          class="absolute w-2 h-2 rounded-full mt-1.5 -start-1 bg-primary"
          data-testid="detail"
        />
        <div
          class="pb-2 ms-4 flex items-center justify-between lg:justify-normal"
        >
          <span
            class="lg:inline-flex items-center border-2 rounded px-2 py-1 font-semibold text-xs text-center border-primary"
            data-testid="time"
          >
            <time
              class="block"
            >
              November 2021
            </time>
          </span>
        </div>
        <div
          class="rounded-xl border bg-card text-card-foreground shadow ms-4"
        >
          <div
            class="flex flex-col space-y-1.5 p-6"
          >
            <div
              class="font-semibold leading-none tracking-tight"
            >
              Certification
            </div>
            <div
              class="text-sm text-muted-foreground"
            >
              Boot-camp Node JS - Ruta N Corporation - Ceiba Software S.A.S - Globant
            </div>
          </div>
          <div
            class="p-6 py-0"
          >
            <div
              class="flex flex-col lg:flex-row"
            />
          </div>
        </div>
      </ol>
      <ol
        class="first-of-type:pt-0 pt-5 relative border-s border-gray-200"
      >
        <div
          class="absolute w-2 h-2 rounded-full mt-1.5 -start-1 bg-primary"
          data-testid="detail"
        />
        <div
          class="pb-2 ms-4 flex items-center justify-between lg:justify-normal"
        >
          <span
            class="lg:inline-flex items-center border-2 rounded px-2 py-1 font-semibold text-xs text-center border-primary"
            data-testid="time"
          >
            <time
              class="block"
            >
              2010
            </time>
            <time
              class="block lg:before:content-['-'] lg:before:mx-1"
            >
              2016
            </time>
          </span>
        </div>
        <div
          class="rounded-xl border bg-card text-card-foreground shadow ms-4"
        >
          <div
            class="flex flex-col space-y-1.5 p-6"
          >
            <div
              class="font-semibold leading-none tracking-tight"
            >
              Professional degree
            </div>
            <div
              class="text-sm text-muted-foreground"
            >
              Software Engineering - Politécnico Colombiano Jaime Isaza Cadavid
            </div>
          </div>
          <div
            class="p-6 py-0"
          >
            <div
              class="flex flex-col lg:flex-row"
            />
          </div>
        </div>
      </ol>
      <ol
        class="first-of-type:pt-0 pt-5 relative border-s border-gray-200"
      >
        <div
          class="absolute w-2 h-2 rounded-full mt-1.5 -start-1 bg-primary"
          data-testid="detail"
        />
        <div
          class="pb-2 ms-4 flex items-center justify-between lg:justify-normal"
        >
          <span
            class="lg:inline-flex items-center border-2 rounded px-2 py-1 font-semibold text-xs text-center border-primary"
            data-testid="time"
          >
            <time
              class="block"
            >
              2009
            </time>
          </span>
        </div>
        <div
          class="rounded-xl border bg-card text-card-foreground shadow ms-4"
        >
          <div
            class="flex flex-col space-y-1.5 p-6"
          >
            <div
              class="font-semibold leading-none tracking-tight"
            >
              Bachelor's degree
            </div>
            <div
              class="text-sm text-muted-foreground"
            >
              Colegio Paula Montal
            </div>
          </div>
          <div
            class="p-6 py-0"
          >
            <div
              class="flex flex-col lg:flex-row"
            />
          </div>
        </div>
      </ol>
    </div>
    <div
      class="basis-full lg:basis-2/3"
    >
      <h2
        class="main-subtitle mt-5 lg:mt-0"
      >
        Languages
      </h2>
      <div
        class="grid grid-cols-2 gap-x-6"
      >
        <div
          class="rounded-xl border bg-card text-card-foreground shadow flex flex-col justify-center items-center"
        >
          <div
            class="flex flex-col space-y-1.5 p-6"
          >
            <svg
              class="lucide lucide-languages text-primary"
              fill="none"
              height="40"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.5"
              viewBox="0 0 24 24"
              width="40"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m5 8 6 6"
              />
              <path
                d="m4 14 6-6 2-3"
              />
              <path
                d="M2 5h12"
              />
              <path
                d="M7 2h1"
              />
              <path
                d="m22 22-5-10-5 10"
              />
              <path
                d="M14 18h6"
              />
            </svg>
          </div>
          <div
            class="p-6 pt-0"
          >
            <p
              class="mt-3 lg:mt-0 text-center text-muted-foreground leading-tight"
            >
              <b>
                Spanish
              </b>
              <small
                class="block mt-1"
              >
                Native (C2)
              </small>
            </p>
          </div>
        </div>
        <div
          class="rounded-xl border bg-card text-card-foreground shadow flex flex-col justify-center items-center"
        >
          <div
            class="flex flex-col space-y-1.5 p-6"
          >
            <svg
              class="lucide lucide-languages text-primary"
              fill="none"
              height="40"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.5"
              viewBox="0 0 24 24"
              width="40"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m5 8 6 6"
              />
              <path
                d="m4 14 6-6 2-3"
              />
              <path
                d="M2 5h12"
              />
              <path
                d="M7 2h1"
              />
              <path
                d="m22 22-5-10-5 10"
              />
              <path
                d="M14 18h6"
              />
            </svg>
          </div>
          <div
            class="p-6 pt-0"
          >
            <p
              class="mt-3 lg:mt-0 text-center text-muted-foreground leading-tight"
            >
              <b>
                English
              </b>
              <small
                class="block mt-1"
              >
                Proficient (B2)
              </small>
            </p>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
